### PR TITLE
Pass proxy address to PromptMFAChallenge calls

### PR DESCRIPTION
**Pull request by codingllama**
Reinstates some logic that was removed on #12475 and changes `optsOverride` to a function, so there is less ambiguity in dealing with booleans / default values.

### CODE REVIEW

File: lib/client/mfa.go [hunk 1]
```diff
@@ -71,23 +71,25 @@ type PromptMFAChallengeOpts struct {
...
 // challenges.
+// If proxyAddr is empty, the TeleportClient.WebProxyAddr is used.
 // See client.PromptMFAChallenge.
 func (tc *TeleportClient) PromptMFAChallenge(
-	ctx context.Context, c *proto.MFAAuthenticateChallenge, optsOverride *PromptMFAChallengeOpts) (*proto.MFAAuthenticateResponse, error) {
+	ctx context.Context, proxyAddr string, c *proto.MFAAuthenticateChallenge,
+	applyOpts func(opts *PromptMFAChallengeOpts)) (*proto.MFAAuthenticateResponse, error) {
+	addr := proxyAddr
+	if addr == "" {
+		addr = tc.WebProxyAddr
```
Comment (marcoandredinis) [inline lib/client/mfa.go:81]: Can we add a test for this one?

**Review by marcoandredinis — APPROVED** [review]
Looks good
Can we add a small test for this one?

A bit of a nitpick but I was expecting to see the `With` pattern that we commonly use with grpc Clients
It would require us to receive a list of `applyOpts` in the PromptMFAChallenge
We could then pass it as many `applyOpts` as we would like

**Comment by codingllama** [comment]
Thanks for the quick review, @marcoandredinis.

We could use the functional options pattern, but I think an opts struct is alright here as well (and a bit simpler). We can always refactor into a more evolved API if necessary.  (TBH I was looking for options to remove while doing this, but sadly we do use them all ATM.)

File: lib/client/mfa.go (hunk 1 above)
Comment (codingllama) [reply lib/client/mfa.go:84]: Done, thanks for calling this out.

**Comment by codingllama** [comment]
PTAL? @marcoandredinis @Joerger @rosstimothy

**Review by rosstimothy — APPROVED** [review]

**Review by marcoandredinis — APPROVED** [review]

**Pull request merged**